Let easyfind client use original easyfind.conf content as state file

gordonb3 opened the pull request:
- First step to allow integration with Bubba web UI

MouettE-SC commented:
Do you really need to use the original path and format for the state file ? This is a state file and as such is genereated and should not be manually edited (/etc is meant for editable configuration files).  I'd prefer to keep it in /var.

gordonb3 commented:
I was unaware of additional pushes to the branch holding my pull request to be auto included. Does one of them move the state file to the /etc tree? Granted: that's where I want it for my Bubba UI project, but it doesn't really need to be for this release. I kind of disagree with the "state only" claim, particularly because the current version does not have any means to request that state unless it already contains a state file that holds the correct name. Strictly you'd need "name" to be in a /etc file and "ip" in /var/lib but I agree that it appears rather pointless to split and divide the working config over two files.

IMO It would be helpful if the new client would behave as closely as possible to the original perl and python based clients, so it can co-operate with them rather than compete. This is important if anyone would want to use this new client/service with the original Bubba OS installation because easyfind is part of the core package bubba-backend.

My local copy is even far more ahead. I have converted the code to C++ to allow additional function entry points. I've also added a program new entry point to return json responses. Which is why I need those additional function entry points - for controlling output in the new entry point without needing to rewrite all existing calls to them.

gordonb3 commented:
Looking back at the code it seems I made something of a mess of this one.

=> revert to main
